src/storage/src/control/status.rs

```diff
@@ -0,0 +1,321 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::google;
+
+impl gaxi::prost::ToProto<google::rpc::Status> for rpc::model::Status {
+    type Output = google::rpc::Status;
+    fn to_proto(self) -> Result<Self::Output, gaxi::prost::ConvertError> {
+        Ok(Self::Output {
+            code: self.code.to_proto()?,
+            message: self.message.to_proto()?,
+            details: self.details.into_iter().filter_map(any_to_prost).collect(),
+        })
+    }
+}
+
+impl gaxi::prost::FromProto<rpc::model::Status> for google::rpc::Status {
+    fn cnv(self) -> Result<rpc::model::Status, gaxi::prost::ConvertError> {
+        Ok(rpc::model::Status::new()
+            .set_code(self.code)
+            .set_message(self.message)
+            .set_details(self.details.into_iter().filter_map(any_from_prost)))
+    }
+}
+
+fn any_to_prost(value: wkt::Any) -> Option<prost_types::Any> {
```

Comment:
You probably noticed this: it /should/ be easy to refactor this function (and its inverse) to `gaxi`, it just requires exporting a function that depends on `prost_types::Any` and `wkt::Any`.

Reply:
But yeah, you are right that we wouldn't have to expose the generated protos in `gaxi` (or some other common crate) to make this work. That is nice.